### DISCUSSION TITLE
`deploy` Documentation Needs More

### DISCUSSION THREAD

**Discussion by shellscape** [pnpm/pnpm#5215] [General]
We've finally upgraded to 7.9 in our work environment and are looking to clean up some scripts that are also named `deploy`. We're also using `make-dedicated-lockfile` for some Docker related deployments. While I'm a contributor and know how `pnpm deploy` works under the hood, the majority of people probably won't. The docs here https://pnpm.io/cli/deploy could be a lot better with regard to:

- what the command actually does; functional steps under the hood
- what files are produced
- where those files are
- what do to with those files during and after deployment, if anything

@zkochan we might also want to consider adding a `Documentation` issue template to the repo

**Comment by josefaidt** [comment]
Came to Discussions to look for existing threads about `deploy`. I use pnpm daily but do not understand what this command is supposed to do other than "deploy a package". I see I can set a flag to install devDependencies or dependencies and specify an output directory, but does `deploy` lean on an existing `deploy` npm script?
